src/components/Dropdowns/AccessibleDropdown.tsx

```diff
@@ -0,0 +1,240 @@
+import { Button, ButtonProps, Dropdown, theme, Tooltip } from "antd";
+import React, { ReactElement, ReactNode, useEffect, useRef, useState } from "react";
+import styled, { css } from "styled-components";
+
+import { DropdownSVG } from "../../assets";
+import { VisuallyHidden } from "../../styles/utils";
+
+type AccessibleDropdownProps = {
+  /** Text label to include with the dropdown. If null or undefined, hides the label. */
+  label?: string | null;
+
+  /** Contents to be rendered inside the dropdown. */
+  dropdownContent: ReactElement;
+  renderDropdownContent?: ((content: ReactElement) => ReactElement) | null;
+  disabled?: boolean;
+  /**
+   * The type of button to render for the dropdown. See Antd's button types:
+   * https://ant.design/components/button#components-button-demo-basic
+   */
+  buttonType?: ButtonProps["type"] | "outlined";
+  buttonText: string;
+  buttonStyle?: React.CSSProperties;
+  /**
+   * Override for the button element. By default, renders `props.buttonText` in a styled Antd button.
+   *
+   * @param props: The props passed to the dropdown component.
+   * @param isOpen: Whether the dropdown is currently open.
+   * @param onClick: The callback to open the dropdown.
+   *
+   * Overrides must call the provided `onClick` callback during click or interaction events for the
+   * dropdown to work correctly.
+   *
+   * Tooltips may not work as expected if the root element is not a button or does not take callbacks
+   *  as props for the following events: `onMouseEnter/Leave`, `onPointerEnter/Leave`,
+   * `onFocus`, or `onBlur`. See implementation of `IconButton.tsx` for an example.
+   */
+  renderButton?: ((props: AccessibleDropdownProps, isOpen: boolean, onClick: () => void) => ReactElement) | null;
```

Comment:
It could... I think I slightly prefer leaving it as a function and not as a component, because then you don't need to create a component every time you want to override this callback.